.github/workflows/test.yml

```diff
@@ -0,0 +1,204 @@
+name: Code Build and Tests
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+permissions:
+  contents: read
+  checks: write
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: Setup dependencies
+        shell: bash
+        run: make download
```

Comment:
Do this instead https://github.com/armadaproject/armada/blob/master/makefile#L576
```
go run github.com/magefile/mage@v1.14.0 BootstrapTools
go mod download
go mod tidy
```

Reply:
So we can start removing the need for the makefile.